pkg/features/kube_features.go

```diff
@@ -409,6 +409,7 @@ var defaultKubernetesFeatureGates = map[utilfeature.Feature]utilfeature.FeatureS
 	genericfeatures.APIResponseCompression:  {Default: false, PreRelease: utilfeature.Alpha},
 	genericfeatures.Initializers:            {Default: false, PreRelease: utilfeature.Alpha},
 	genericfeatures.APIListChunking:         {Default: true, PreRelease: utilfeature.Beta},
+	genericfeatures.DryRun:                  {Default: false, PreRelease: utilfeature.Alpha},
```

Comment:
I expected this to be set for struct initialized by the generic apiserver options, not here.

Reply:
I think one negative of doing it that way is that the list of features/defaults would be split between multiple locations

Reply:
It's already set in two places @jennybuckley, and I think what @deads2k is saying is that it should only be set in one place (the one in staging/src/k8s.io/apiserver/pkg/features/kube_features.go).

Reply:
Oh, I see that makes sense. Don't we need these in both then?

Reply:
I don't see why we'd need to add DryRun here, that's just increasing the leak.

Reply:
We have it at two places intentionally. But there is code to detect when it diverges.

The reason to have all of them here is to have an overview over all existing feature flags.

I would not worry about this for the moment. We are going to refactor this structure to avoid leakage of the gates into other, unintended components. But we don't have to solve this here.